third_party/amd/lib/TritonAMDGPUTransforms/MfmaGroup.cpp

```diff
@@ -258,12 +340,26 @@ std::pair<mlir::Type, mlir::Type> TypesFromMfmaId(mlir::MLIRContext *ctx,
 }
 
 FailureOr<MfmaInsn> MfmaInsn::selectMfma(unsigned mDim, unsigned nDim,
-                                         Type elementTypeA, Type elementTypeB,
-                                         int mfmaVersion, bool allowXF32) {
+                                         unsigned kDim, Type elementTypeA,
+                                         Type elementTypeB, int mfmaVersion,
+                                         bool allowXF32) {
   auto mfmaInsnAttrMap = getMfmaInsnGroupAttrMap();
   MfmaTypeId mfmaId =
       chooseAppropriateMfmaId(elementTypeA, elementTypeB, allowXF32);
-  MfmaInsnGroupSelectKey key = {mDim, nDim, mfmaId, mfmaVersion};
+  // gfx950: select double-rate mfma ops for large kDim
+  if (mfmaVersion == 4 &&
+      (mfmaId == MfmaTypeId::Fp16TyId || mfmaId == MfmaTypeId::Bf16TyId)) {
+    if (mDim == 16 && nDim == 16 && kDim >= 32)
+      kDim = 32;
+    else if (mDim == 32 && nDim == 32 && kDim >= 16)
+      kDim = 16;
+    else
+      kDim = 0;
+  }
+  // non-gfx950: ignore kDim
+  else
+    kDim = 0;
```

Comment:
We are modifying the `kDim`; it can be confusing. Can you create a `usigned kDimKey = 0` local variable and assign to that? This way you also don't need to have the `kDim = 0` assignment anymore.